doc/instruction_decode_execute.rst

```diff
@@ -0,0 +1,84 @@
+.. _instruction-decode-execute:
+
+Instruction Decode and Execute
+==============================
+
+.. figure:: images/de_ex_stage.svg
+   :name: de_ex_stage
+
+   Instruction Decode and Execute
+
+The Instruction Decode and Execute stage takes instruction data from the instruction fetch stage (which has been converted to the uncompressed representation in the compressed instruction case). The instructions are decoded and executed all within one cycle including the register read and write. The stage is made up of multiple sub-blocks which are described below.
+
+ID Stage
```

Comment:
I think it's confusing to call this "ID Stage" when you've already described decode+execute happening in a single stage.